ide/che-core-ide-api/src/main/java/org/eclipse/che/ide/api/machine/WsAgentStateController.java

```diff
@@ -151,63 +188,92 @@ public void makeCall(AsyncCallback<MessageBus> callback) {
+                    for (int i = 0; i < rootResources.size(); i++) {
+                        JSONObject rootResource = rootResources.get(i).isObject();
+                        String regex = rootResource.get("regex").isString().stringValue();
+                        String fqn = rootResource.get("fqn").isString().stringValue();
+                        String path = rootResource.get("path").isString().stringValue();
+                        availableServices.add(new RestServiceInfo(fqn, regex, path));
+                    }
+                }
+
+                checkWsConnection();
+            }
+        }).catchError(new Operation<PromiseError>() {
+            @Override
+            public void apply(PromiseError arg) throws OperationException {
+                asyncRequestFactory.createGetRequest(pingUrl)
+                                   .header(ACCEPT, APPLICATION_JSON)
+                                   .send(dtoUnmarshallerFactory.newUnmarshaller(AgentHealthStateDto.class))
+                                   .then(new Operation<AgentHealthStateDto>() {
                                        @Override
-                                       public void run() {
-                                           checkHttpConnection();
+                                       public void apply(AgentHealthStateDto arg) throws OperationException {
+                                           if (RUNNING.equals(arg.getWorkspaceStatus())) {
+                                               checkStateOfWsAgent(arg);
+                                           }
                                        }
-                                   }.schedule(1000);
-                               }
-                           });
+                                   }).catchError(new Operation<PromiseError>() {
+                    @Override
+                    public void apply(PromiseError arg) throws OperationException {
+                        Log.error(getClass(), arg.getMessage());
+                    }
+                });
+            }
+        });
+    }
+
+    private void checkStateOfWsAgent(AgentHealthStateDto agentHealthStateDto) {
+        if (!WSAGENT_REFERENCE.equals(agentHealthStateDto.getAgentId())) {
+            return;
+        }
+
+        final AgentStateDto agentState = agentHealthStateDto.getAgentState();
+        if (agentState == null) {
+            return;
+        }
+
+        final int statusCode = agentState.getCode();
+
+        String infoWindowTitle = "Workspace Agent";
+
+        if (statusCode == 200) {
+            dialogFactory.createMessageDialog(infoWindowTitle,
+                                              "Your workspace is not responding. To fix the problem, verify you have a good " +
+                                              "network connection and restart the workspace.",
+                                              new ConfirmCallback() {
```

Comment:
ConfirmCallback is nullable so no need to create this ConfirmCallback that does nothing